integration_tests/tests/idf_station.rs

```diff
@@ -0,0 +1,173 @@
+                &[
+                    &station,
+                    &seasons,
+                    &qc,
+                    &seed_parameter,
+                    &fromtime,
+                    &totime,
+                    &updated_at,
+                ],
+            )
+            .await
+            .unwrap()
+            .get(0);
+
+        for duration in durations.iter() {
+            for freq in frequencies.iter() {
+                for val in intensity.iter() {
+                    conn.execute(&data_query, &[&tsid, duration, freq, val, &lower, &upper])
+                        .await
+                        .unwrap();
+                }
+            }
+        }
+    }
+
+    TestData {
+        stations,
+        // durations,
+        // frequencies,
+        intensity,
+        lower,
+        upper,
+    }
+}
+
+#[tokio::test]
+async fn test_idf_station_availability() {
+    e2e_test_wrapper(async {
+        let expected = TEST_DATA.get_or_init(init_idf_data).await;
+
+        // Returns results sorted by station ID
+        let url = "http://localhost:3000/reports/idf/station";
+
+        let resp = reqwest::get(url).await.unwrap();
+        assert!(resp.status().is_success());
+
+        let resp: IdfStationAvailability = resp.json().await.unwrap();
+        assert_eq!(resp.stations.len(), expected.stations.len());
+
+        for (station, exp) in resp.stations.iter().zip(expected.stations.iter()) {
+            assert_eq!(station.station_id, *exp)
+        }
+    })
+    .await
+}
+
+#[tokio::test]
+async fn test_idf_station_endpoint() {
+    e2e_test_wrapper(async {
+        let data = TEST_DATA.get_or_init(init_idf_data).await;
+
+        let station_id = data.stations[0];
+        let url = format!("http://localhost:3000/reports/idf/station/{}", station_id);
+
+        let resp = reqwest::get(url).await.unwrap();
+        assert!(resp.status().is_success());
+
+        let resp: IdfStationResp = resp.json().await.unwrap();
+        assert_eq!(resp.station_id, station_id);
+    })
+    .await
+}
+
+#[tokio::test]
+async fn test_idf_station_endpoint_with_unit() {
+    e2e_test_wrapper(async {
+        let data = TEST_DATA.get_or_init(init_idf_data).await;
+
+        let station_id = data.stations[1];
+        let url = format!(
+            "http://localhost:3000/reports/idf/station/{}?unit=lsha",
+            station_id,
+        );
+
+        let resp = reqwest::get(url).await.unwrap();
+        assert!(resp.status().is_success());
+
+        let resp: IdfStationResp = resp.json().await.unwrap();
+        assert_eq!(resp.station_id, station_id);
+
+        for val in resp.values {
+            // Since the values get converted from 'mm' to 'lsha'
+            // we should not find them in the response
+            assert!(!data.intensity.contains(&val.intensity));
+            assert_ne!(val.lower_interval, data.lower);
+            assert_ne!(val.upper_interval, data.upper);
+        }
+    })
+    .await
+}
```

Comment:
Would be good to check that the data is correct not just that it's there

Reply:
Done, not sure it's the ideal way though